### DISCUSSION TITLE
DatePicker - SelectedDateChanged Event

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#12409] [Q&A] [answered]
Hello,

I have a DatePicker in a view and want to update the view when DatePickers SelectedDate has changed.

Using SelectedDateChanged event does not do a good job since it is fired before the SelectedDate has been synchronized back to the binded property. 

```
    private void DatePicker_OnSelectedDateChanged(object? sender, DatePickerSelectedValueChangedEventArgs e)
    {
#if _LOCALDEBUG_
        Logger?.Debug(this, $"{ViewModel.Von} -> {ViewModel.Bis}");
        Logger?.Debug(this, $"{e.OldDate}");
        Logger?.Debug(this, $"{e.NewDate}");
#endif
        ViewModel.Refresh();
    }
```

My refresh methode (reloading data from database) takes the unchanged property since the properties _Von_ or _Bis_ are not already synced back. I would expect such behavior for an event called **SelectedDateChanging** since **SelectedDateChanged** implies the change is already done and not still in progress.

But nevertheless, how is it intended to handle actions which have to take place after the change is finished and all binded properties are synced?

Joachim

**Comment by timunie** [comment, marked as answer]
Hello Joachim, 

why don't you handle this inside your ViewModel? if it implements `INPC`, then you can listen to it's property changes and update the collection accordingly. 

BR 
Tim

**Reply by drei97** [reply]
Good question!
